<!DOCTYPE html>
<html>

<head>
  <meta http-equiv="content-type" content="text/html; charset=utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="apple-mobile-web-app-capable" content="yes" />
<title>3D 基因组与生物信息 | 思考问题的熊</title>
<link rel="shortcut icon" href="https://kaopubear.top/blog/favicon.ico">
<link href="https://kaopubear.top/blog/styles/main.css?v=1723040174420" rel="stylesheet">
<link href="//at.alicdn.com/t/font_1678829_1mxi55l93a3.css" rel="stylesheet">
<link rel="alternate" type="application/rss+xml" title="思考问题的熊 » Feed"
  href="https://kaopubear.top/blog/atom.xml">


<script async src="https://www.googletagmanager.com/gtag/js?id=UA-91022805-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];

  function gtag() {
    dataLayer.push(arguments);
  }
  gtag('js', new Date());

  gtag('config', 'UA-91022805-1');
</script>


  <meta name="description" content="重要的 Hi-C 相关文献
第一篇 Hi-C 文章： Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Geno..." />
  <meta name="keywords" content="生物信息" />
</head>

<body>
  <div class="main animated">
    <div class="header animated fadeInDown">
  <div class="site_title_container">
    <div class="site_title">
      <a href="https://kaopubear.top/blog">
        思考问题的熊
      </a>
    </div>
    <div class="description">
      很高兴和你相遇<br>这里正在记录我的所思所学<br>订阅<a href="https://kaopubear.zhubai.love/" target="_blank">免费邮件通讯</a>接收最新内容
    </div>
  </div>
  <div class="my_socials">
    
    
    <a href="https://github.com/fei0810" target="_blank">
      <i class="iconfont icon-github"></i></a>
    
          
    
    <a href="https://twitter.com/kaopubear" target="_blank">
      <i class="iconfont icon-twitter"></i></a>
    
          
    
          
    
          
    
    <a href="https://t.me/kaopubear" target="_blank">
      <i class="iconfont icon-telegram"></i></a>
    
          
    <a href="https://kaopubear.top/blog/atom.xml" target="_blank">
      <i class="iconfont icon-rss"></i></a>
  </div>
</div>

<div class="header_menu">
  
  
  <a href="https://kaopubear.top/blog" class="menu">首页</a>
  

  
  <a href="https://kaopubear.top/blog/archives" class="menu">归档</a>
  

  
  <a href="https://talk.kaopubear.top/" class="menu" target="_blank">
    想法</a>
  

  
  <a href="https://kaopubear.top/blog/sharelink" class="menu">工具</a>
  

  
  <a href="https://kaopubear.zhubai.love/" class="menu" target="_blank">
    通讯</a>
  

  
  <a href="https://kaopubear.podcast.xyz/" class="menu" target="_blank">
    播客</a>
  

  
  <a href="https://kaopubear.top/resume/" class="menu" target="_blank">
    简历</a>
  

  
  <a href="https://kaopubear.top" class="menu" target="_blank">
    主页</a>
  

  <div class="gridea-search-div">
    <form id="gridea-search-form" data-update="1723040174420" action="https://kaopubear.top/blog/search/">
      <input class="gridea-search-input" autocomplete="off" spellcheck="false" name="q" />
    </form>
  </div>
</div>

    <div class="autopagerize_page_element">
      <div class="content">
        <div class="post_page">
          <div class="post animated fadeInDown">
            <div class="post_title post_detail_title">
              <h2>
                <a>
                  3D 基因组与生物信息</a>
              </h2>
              <span class="article-info">2018-08-11, 854 words, 5 min read</span>
            </div>
            <div class="post_content markdown">
              <p class="md_block">
                <span class="md_line md_line_start md_line_end">
                  <h2 id="重要的-hi-c-相关文献">重要的 Hi-C 相关文献</h2>
<p>第一篇 Hi-C 文章： Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome； DOI: 10.1126/science.1181369</p>
<!--more-->
<p>TAD 提出： Topological Domains in Mammalian Genomes Identified by Analysis of Chromatin Interactions ；doi: 10.1038/nature11082</p>
<p>高分辨率 Hi-C： A 3D Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping <a href="https://doi.org/10.1016/j.cell.2014.11.021">https://doi.org/10.1016/j.cell.2014.11.021</a></p>
<p>单细胞 Single cell： Hi-C reveals cell-to-cell variability in chromosome structure doi: 10.1038/nature12593；3D structures of individual mammalian genomes studied by single-cell Hi-C doi:10.1038/nature21429</p>
<p>综述：老师所讲 Hi-C 相关基础知识主要来自于综述 Organization and function of the 3D genome，doi:10.1038/nrg.2016.112</p>
<h2 id="chromatin-interaction-in-different-resolutions">Chromatin interaction in different resolutions</h2>
<p>不同分辨率 Hi-C 可以看到的内容不同</p>
<p>5KB 可以看到各种 loop</p>
<figure data-type="image" tabindex="1"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-828599c54149be5d.jpg" alt="" loading="lazy"></figure>
<p>10KB 可以看到 TAD</p>
<figure data-type="image" tabindex="2"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-02e99cd9e50fa311.jpg" alt="" loading="lazy"></figure>
<p>50kb 可以看到 TAD 之间的关联</p>
<figure data-type="image" tabindex="3"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-44370d2fbc86ef43.jpg" alt="" loading="lazy"></figure>
<p>在整个染色体的水平可以看到染色质的位置分布</p>
<figure data-type="image" tabindex="4"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-c2baa72eda2f9bb2.jpg" alt="" loading="lazy"></figure>
<h2 id="什么造成了所谓的-tad">什么造成了所谓的 TAD</h2>
<figure data-type="image" tabindex="5"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/e556f4a1-7f13-4a38-8943-9dc29cfb9314.jpg" alt="" loading="lazy"></figure>
<h3 id="cohesin-complex">cohesin complex</h3>
<p>Cohesin is a protein complex that regulates the separation of sister chromatids during cell division, either mitosis or meiosis.</p>
<p>Cohesins hold sister chromatids together after DNA replication until anaphase when removal of cohesin leads to separation of sister chromatids.</p>
<h3 id="ctcf-proteins">CTCF proteins</h3>
<p>转录阻抑物 CTCF</p>
<p>CTCF 与靶顺序因子的结合可阻断增强子和启动子的相互作用，从而将增强子的活性限制在一定的功能区域</p>
<p>除了阻断增强子外，CTCF 还可作为染色质屏障阻止异染色质的传播</p>
<h2 id="predicting-enhancer-promoter-loops-如何预测-epl">Predicting enhancer-promoter loops 如何预测 EPL</h2>
<h3 id="两种类似的算法">两种类似的算法</h3>
<p>TargetFinder(Whalen et al. Nat Gen 2016)— an algorithm that uses many functional genomic datasets, including DNase-seq, histone marks, transcription factor (TF) ChIP-seq, gene expression, and DNA methylation data etc.</p>
<ul>
<li>
<p>Enhancer–promoter interactions are encoded by complex genomic signatures on looping chromatin ,doi:10.1038/ng.3539</p>
</li>
<li>
<p>pipeline</p>
</li>
</ul>
<figure data-type="image" tabindex="6"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-368541bebfbdc802.jpg" alt="" loading="lazy"></figure>
<p>RIPPLE (Roy et al. NAR 2016) — Also uses functional genomic datasets for feature extraction.</p>
<ul>
<li>A predictive modeling approach for cell line-specific long-range regulatory interactions , <a href="https://doi.org/10.1093/nar/gkv865">https://doi.org/10.1093/nar/gkv865</a></li>
</ul>
<p>二者共同的发现</p>
<ul>
<li>signals from these functional genomic data are informative to computationally distinguish enhancer-promoter interactions from noninteracting enhancer-promoter pairs.</li>
</ul>
<h3 id="pep-只用序列信息来进行分析马坚实验室">PEP 只用序列信息来进行分析（马坚实验室）</h3>
<figure data-type="image" tabindex="7"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-eb196cce2ead61a9.jpg" alt="" loading="lazy"></figure>
<h2 id="hi-c-分析流程">Hi-C 分析流程</h2>
<p>Analysis methods for studying the 3D architecture of the genome ，<a href="https://doi.org/10.1186/s13059-015-0745-7">https://doi.org/10.1186/s13059-015-0745-7</a></p>
<p>流程</p>
<figure data-type="image" tabindex="8"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-7df3e724f72f8706.jpg" alt="" loading="lazy"></figure>
<h2 id="contact-map">contact map</h2>
<p>定义：A contact map is a matrix with rows and columns representing non-overlapping ‘bins’ across the genome.</p>
<p>Each entry in the matrix contains a count of read pairs that connect the corresponding bin pair in a Hi-C experiment.</p>
<p>How to determine bin size</p>
<ul>
<li>No standard rule. Rao et al. 2014 suggests using a bin size that results in at least 80% of all possible bins with &gt;1000 contacts.</li>
</ul>
<p>Two types of approaches to correct bias in the contact map</p>
<ul>
<li>
<p>Explicit approach — assuming some known bias</p>
<pre><code>- Restriction enzyme fragment lengths, GC content, and sequence mappability are three major sources of biases in Hi-C data (Yaffe and Tanay, Nat Genet 2011)
- HiCNorm — simpler and faster (Hu et al. Bioinformatics 2012)
</code></pre>
</li>
<li>
<p>Implicit approach — assume no known source of bias and that each locus receives equal sequence coverage after biases are removed</p>
<pre><code>- In other words, if there is no bias, the total genome-wide contact summation for each locus will be a constant, i.e., each locus has 'equal visibility'
</code></pre>
</li>
</ul>
<h2 id="contact-matrix-normalization">Contact matrix normalization</h2>
<p>如何进行标准化</p>
<figure data-type="image" tabindex="9"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-8e50a48b35e02af0.jpg" alt="" loading="lazy"></figure>
<h2 id="鉴别-tad-的算法">鉴别 TAD 的算法</h2>
<p><strong>HMM（任兵）</strong></p>
<figure data-type="image" tabindex="10"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-ee87211eb4b5a3fd.jpg" alt="" loading="lazy"></figure>
<p><strong>Arrowhead</strong></p>
<figure data-type="image" tabindex="11"><img src="https://kaopubear-1254299507.file.myqcloud.com/picgo/177622-53bc76992ca74202.jpg" alt="" loading="lazy"></figure>
<hr>
<p><strong>本文作者</strong>：思考问题的熊</p>
<p><strong>版权声明</strong>：本博客所有文章除特别声明外，均采用 <a href="http://creativecommons.org/licenses/by-nc-nd/4.0">知识共享署名-非商业性使用-禁止演绎 4.0 国际许可协议 (CC BY-NC-ND 4.0)</a> 进行许可。</p>
<p>如果你对这篇文章感兴趣，欢迎通过邮箱或者微信订阅我的 <strong>「熊言熊语」会员通讯</strong>，我将第一时间与你分享<strong>肿瘤生物医药领域最新行业研究进展</strong>和我的<strong>所思所学所想</strong>，<a href="https://kaopubear.zhubai.love/">点此链接</a>即可进行免费订阅。<img src="https://kaopubear-1254299507.cos.ap-shanghai.myqcloud.com/picgo/202212100024561.webp" alt="" loading="lazy"></p>
<br />
                  
                  <span class="copyright">· 分享链接 https://kaopubear.top/blog/2017-08-11-longxing-bioinfo-hic/ </span>
                </span>
                
              </p>
            </div>
            <div class="post_footer">
              
                <div class="meta">
                  <div class="info">
                    <span class="field tags">
                      <i class="iconfont icon-tag-sm"></i>
                      
                      <a href="https://kaopubear.top/blog/DvI6ypPftI/" class="article-info">
                        生物信息</a>
                      </span>
                  </div>
                </div>
                
              
                <div class="next-post" style="margin-top: 20px;">
                  <div class="next">下一篇</div>
                  <a href="https://kaopubear.top/blog/2018-08-05-ecluseevernote/">
                    <h3 class="post-title">
                      一位印象大使的印象笔记使用心得
                    </h3>
                  </a>
                </div>
                
            </div>
          </div>
          
          
                      
          <link rel="stylesheet" href="https://unpkg.com/disqusjs@1.1/dist/disqusjs.css">
<script src="https://unpkg.com/disqusjs@1.1/dist/disqus.js"></script>
<div id="disqus_thread"></div>
<script>
  var options = {
    shortname: 'kaopubear',
    apikey: 'S3aqfVwpSDyKQ2A8Zpg9bXlFmZue0eFVbb3JWkNZtefxqCH39Fd2cLkwP7X0MPDi',
  }
  if ('') {
    options.api = ''
  }
  var dsqjs = new DisqusJS(options)
</script>
          
                            
        </div>
      </div>
    </div>
  </div>
  <div class="footer">
  
  <div class="powered">
    Powered by<a href="https://gridea.dev/" target="_blank">Gridea</a>&<a
    href="https://github.com/fei0810/gridea-theme-oneplus" target="_blank">Theme OnePlus</a>
 </div>
<div class="footer_slogan">
  
</div>
<div id="back_to_top" class="back_to_top">
  <span>△</span>
</div>
<script src="https://kaopubear.top/blog/media/scripts/util.js"></script>

</div>
  <link rel="stylesheet" href="//cdn.jsdelivr.net/gh/highlightjs/cdn-release/build/styles/default.min.css">
  <script src="//cdn.jsdelivr.net/gh/highlightjs/cdn-release/build/highlight.min.js"></script>
  <script>
    hljs.initHighlightingOnLoad();
  </script>
</body>

</html>